keep/api/routes/providers.py

```
@@ -607,22 +607,37 @@ def invoke_provider_method(
+            select(Provider).where(
+                (Provider.tenant_id == tenant_id) & (Provider.id == provider_id)
+            )
+        ).one()
+        provider_id = provider.id
+        provider_type = provider.type
+        provider_config = secret_manager.read_secret(
+            provider.configuration_key, is_json=True
+        )
+    except NoResultFound:
+        if not provider_id.startswith("default-"):
+            raise HTTPException(404, detail="Provider not found")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)